rdmd.d

```diff
@@ -114,6 +114,14 @@ int main(string[] args)
     assert(programPos > 0);
     auto argsBeforeProgram = args[0 .. programPos];
 
+    /* Catch -main and handle it like --main. This needs to be done, because
+    rdmd compiles the root file independently from the dependencies, but -main
+    must be present in only one of the calls to dmd. */
+    foreach (ref arg; argsBeforeProgram)
+    {
+        if (arg == "-main") arg = "--main";
```

Comment:
I really like that trick :+1:  (I filtered for it for the additional source compilation)